### ISSUE TITLE
Throwing exceptions wastes resources

### ISSUE THREAD

**Issue by ByronAP** [justcoding121/Titanium-Web-Proxy#634]
It looks like there are a few places where exceptions are intentionally being throw. This seems to have a very adverse effect on performance, especially in RequestHandler and how they bubble up.

**Comment by honfika** [comment]
Agree, I also hate that much exceptions. Something shoud be done :)

**Comment by justcoding121** [comment]
I noted up to 54 exception explicitly thrown. I am to blame for writing the most new exceptions :-)
[https://github.com/justcoding121/Titanium-Web-Proxy/search?q=throw](https://github.com/justcoding121/Titanium-Web-Proxy/search?q=throw)

Some of the exceptions are thrown so that the client/server connection can be disposed on the parent call. A way to reduce this would be to log the error, then dispose the connections immediately if they are in scope. If client/server connections are in parent scope, we need to return an error status from child method to parent method, so that parent can call dispose.

**Comment by justcoding121** [comment]
﻿Implemented a focused reduction of exception-driven control flow (commit 8323e1c3).

**What changed**
- User TerminateSession / cancellation now throws OperationCanceledException instead of a generic Exception("Session was terminated by user.") in RequestHandler, ExplicitClientHandler, TransparentClientHandler, and Http11ToHttp2BridgeHandler.
- RequestHandler (and the HTTP/1.1-to-HTTP/2 bridge) no longer wrap OperationCanceledException or RetryableServerConnectionException into ProxyHttpException just for logging/propagation.
- Explicit/Transparent client handlers treat cancellation as benign at the boundary (no Error-level wrap).
- ProxyDiagnostics.IsExpected treats OCE, IO/socket disconnects, timeouts, and RetryableServerConnectionException as expected/benign.
- ReadResponseStatus returns null on EOF/empty close (no throw); callers map that to retryable stale-connection failure when safe, or IOException when a response was required.
- Characterization tests cover keep-alive happy path (no first-chance proxy throws), TerminateSession (AfterResponse/dispose still runs, not logged as Error), client disconnect, diagnostics classification, and strict malformed status lines.

**Intentionally left as exceptions**
- Malformed protocol / framing (invalid status lines, chunk lengths, HTTP/2 preface violations, etc.).
- Public typed proxy exceptions (ProxyHttpException, ProxyConnectException, ProxyTimeoutException, etc.) for caller-visible failures.
- RetryableServerConnectionException itself (still the retry signal before body consumption; not rewritten to a full result-type RetryPolicy in this PR).
- Genuine I/O failures after a streamed request body (cannot safely retry).
- Prefetch miss remains a null result (unchanged).
